[jit] cleanups to alias analysis

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21227 [jit] make analyzeConservative more conservative
* #21224 [jit] cleanup templated implementation of mayAlias
* #21223 [jit] use memory locations instead of values for working set
* #21222 [jit] remove multisets from work set
* **#21221 [jit] cleanups to alias analysis**

1. AT_ASSERT replaced with TORCH_INTERNAL_ASSERT.
2. Tighten restrictions on where wildcards can appear.

Differential Revision: [D15592313](https://our.internmc.facebook.com/intern/diff/D15592313)

suo committed:
[jit] cleanups to alias analysis

1. AT_ASSERT replaced with TORCH_INTERNAL_ASSERT.
2. Tighten restrictions on where wildcards can appear.

zdevito commented on torch/csrc/jit/passes/alias_analysis.cpp:
```diff
@@ -338,23 +338,7 @@ void AliasDb::analyzeImpl(Node* node) {
       if (tryRegisteredAnalysis(node)) {
         return;
       }
-      AT_ASSERT(!aliasAnalysisHasSpecialCaseFor(node->kind()));
```
This stuff isn't needed anymore?

suo committed:
Update on "[jit] cleanups to alias analysis"

[jit] cleanups to alias analysis

1. AT_ASSERT replaced with TORCH_INTERNAL_ASSERT.
2. Tighten restrictions on where wildcards can appear.

gh-metadata: pytorch pytorch 21221 gh/suo/45/head

suo committed:
Update on "[jit] cleanups to alias analysis"

[jit] cleanups to alias analysis

1. AT_ASSERT replaced with TORCH_INTERNAL_ASSERT.
2. Tighten restrictions on where wildcards can appear.

gh-metadata: pytorch pytorch 21221 gh/suo/45/head

suo committed:
Update on "[jit] cleanups to alias analysis"

[jit] cleanups to alias analysis

1. AT_ASSERT replaced with TORCH_INTERNAL_ASSERT.
2. Tighten restrictions on where wildcards can appear.

gh-metadata: pytorch pytorch 21221 gh/suo/45/head

suo committed:
Update on "[jit] cleanups to alias analysis"

[jit] cleanups to alias analysis

1. AT_ASSERT replaced with TORCH_INTERNAL_ASSERT.
2. Tighten restrictions on where wildcards can appear.

gh-metadata: pytorch pytorch 21221 gh/suo/45/head
